src/testController.ts

```diff
@@ -0,0 +1,197 @@
+import { exec } from "child_process";
+import { promisify } from "util";
+
+import * as vscode from "vscode";
+import { CodeLens } from "vscode-languageclient/node";
+
+import { Ruby } from "./ruby";
+import { Command } from "./status";
+
+const asyncExec = promisify(exec);
+
+export class TestController {
+  private testController: vscode.TestController;
+  private testCommands: WeakMap<vscode.TestItem, string>;
+  private testRunProfile: vscode.TestRunProfile;
+  private workingFolder: string;
+  private ruby: Ruby;
+
+  constructor(
+    context: vscode.ExtensionContext,
+    workingFolder: string,
+    ruby: Ruby
+  ) {
+    this.workingFolder = workingFolder;
+    this.ruby = ruby;
+
+    this.testController = vscode.tests.createTestController(
+      "rubyTests",
+      "Ruby Tests"
+    );
+
+    this.testCommands = new WeakMap<vscode.TestItem, string>();
+
+    this.testRunProfile = this.testController.createRunProfile(
```

Comment:
Can we have more than 1 profile defined? Could we also have a run profile that enables debugging? (Probably not as part of this PR, though)

Reply:
Yeah, we intend to create a debugging profile to launch debugging sessions from tests.